### ISSUE TITLE
Datagrid: requesting a property to control whether to empty the column cell when opening a custom editor

### ISSUE THREAD

**Issue by claudenbach** [infor-design/enterprise#5849]
**Is your feature request related to a problem or use case? Please describe.**
Landmark has the concept of a code block column.  For a code block column we have developed a custom editor. When the code block cell has focus, we are opening a modal dialog to fill out the fields that make up a code block.  

Within the method makeCellEditable(), it always executes cellNode.empty(). This blanks out the current value in the cell.

**Describe the solution you'd like**
Provide an additional property on the column, that when present the cellNode.empty() is not performed.

Perhaps something like: column.doNotEmptyCellWhenEditing: boolean 

So the code would be something like...
```

    // In Show Editor mode the editor is on form already
    if (!col.inlineEditor) {
      if (isEditor) {
        cellNode.css({ position: 'static', height: cellNode.outerHeight() });
      }
      cellParent.addClass('is-editing');

      if (!col.doNotEmptyCellWhenEditing) {
        cellNode.empty();
      }
    } else {
      cellParent.addClass('is-editing-inline');
    }

```
**Describe alternatives you've considered**
Open for other alternatives for solving.

**Additional context**
What ever property is created to handle, need to expose it in enterprise-ng for usage in Landmark AG.

**Comment by claudenbach** [comment]
Here is the landmark example:

![Git-5849](https://user-images.githubusercontent.com/39804038/141211200-ebdc57ab-606d-4cdf-83c0-7460d699b042.gif)

**Comment by tmcconechy** [comment]
We can investigate but im wondering why we empty the cell. If you enter a cell on here https://main-enterprise.demo.design.infor.com/components/datagrid/example-editable.html it just edits the current value

**Comment by tmcconechy** [comment]
Questioning if we should even clear the cell contents at all... We should review that

**Comment by tjamesallen15** [comment]
> Questioning if we should even clear the cell contents at all... We should review that

Per my investigation, we can't just remove it. Please see my pm. For now I would just follow through what the user wants. Although I don't have any example pages for this one since they're using custom ones.

**Comment by tmcconechy** [comment]
@claudenbach can you assist with a way for @tjamesallen15 to reproduce.
One thing we could just do is add the setting as suggested in the code comments? So its on a setting

**Comment by CindyMercadoReyes** [comment]
This issue is now resolved.

**Comment by claudenbach** [comment]
Testing with ids-enterprise-ng 10.13.0, I didn't see were the new column property of doNotEmptyCellWhenEditing was exposed for use in NG?  

Change needed to soho-datagrid.d.ts...

doNotEmptyCellWhenEditing?: boolean;

Do I now need to create a enterprise-ng ticket?  Change will need to go into 10.13.x.

**Comment by tmcconechy** [comment]
@claudenbach if you want to PR the type. When merged i can add to 10.13.x for you.

**Comment by claudenbach** [comment]
@tmcconechy I'm unable to complete, getting a permission to infor-design/enterprise-ng.git denied. 

I'm logged into github.com so I'm unable to proceed.

Need to add to the interface SohoDataGridColumn within soho-datagrid.d.ts the following property:

  // Do not blank out the cell when editing
  doNotEmptyCellWhenEditing?: boolean;

I placed this property right after the editorOptions?: any.

**Comment by tmcconechy** [comment]
Tried to add you again but you were in the permission group. What steps did you take and what error?

I can add that quick if you want but just trying to solve for future.. This is what i do:

```
cd my-folder
git clone git@github.com:infor-design/enterprise-ng.git .
npm i
git checkout -b my-branch
// do changes
git commit -a
git push
```

**Comment by tmcconechy** [comment]
ok, then i dont know about that but basically
-checkout
-branch
-fix
-commit
-push
- p

Whatever they call it

**Comment by tmcconechy** [comment]
OK added and patched to 10.13.1 (ids-enterprise-ng)

**Comment by claudenbach** [comment]
Thank you!
